### PR TITLE
A GetCoordinates dispatch function for kernels templated on pack type

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [x] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by Yurlungur** [comment]
Oops forgot to write what this is above:

## PR description

I got tired of jumping through hoops to access `Coordinates_t` in generic templated functions that are intended to operate on either `MeshData` or `MeshBlockData`, so I wrote a function to jump through those hoops for me. I chose this approach over carrying the coordinates object into the `VariablePack` to reduce the amount of data we're copying around and to keep the `VariablePack`s lightweight.

An alternative would be to put `coords` into the `VariablePack` and redesign `MeshBlockPack` to expose it, rather than carry around its own array of the objects. I'd be fine either way, but I chose this method as a first pass, since it seems easy and lightweight.

Pinging folks who I think would be interested, as well as downstream developers who may have run into this. @jdolence @brryan @lroberts36 @pgrete @forrestglines

**Comment by Yurlungur** [comment]
@par-hermes format

**Comment by Yurlungur** [comment]
@par_hermes format

**Comment by Yurlungur** [comment]
@par-hermes format

**Comment by Yurlungur** [comment]
@par-hermes format

**Comment by Yurlungur** [comment]
@pgrete @jdolence @lroberts36 @forrestglines any opinions on this?

File: docs/interface/containers.md [hunk 1]
```diff
@@ -233,3 +233,25 @@ MeshBlockVarFluxPack<T>
...
 which correspond to packs over `MeshBlock`s that contain just variables
 or contain variables and fluxes.
+
+## Packs and Coordinates
+
+The coordinate system for a given `MeshBlock` object is contained in
+`MeshBlock.coords`, a public field. As discussed above,
+`MeshBlockPack` objects contain a 1D array of `coords` objects. The
+difference between these two interfaces... a Kokkos view in one case,
+and a single static object in the other, can cause difficulties when
```
Comment (lroberts36) [inline docs/interface/containers.md]: Maybe replace `... a Kokkos view in one case, and a single static object in the other,` with `-- a Kokkos view in one case, and a single static object in the other --`. The meaning of the ellipsis was confusing to me when I first read it.

File: src/coordinates/get_coordinates.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace parthenon {
+
+template <>
+ParArray1D<Coordinates_t> GetCoordinates(MeshBlockData<Real> *rc, VariablePack<Real> &p) {
+  ParArray1D<Coordinates_t> coords("Parthenon::Coordinates_t array", 1);
+  auto pmb = rc->GetParentPointer();
+  auto pmb_coords = pmb->coords;
+  pmb->par_for(
+      "fill coords view", 0, 0, KOKKOS_LAMBDA(const int i) { coords(0) = pmb_coords; });
```
Comment (lroberts36) [inline src/coordinates/get_coordinates.cpp:28]: Is it likely to be a performance issue that there is a kernel launch every time this function is called?

**Review by lroberts36 — COMMENTED** [review]
This looks like a reasonable approach to me. Another approach might be to add `ParArray1D<Coordinates_t>& coord_array = coords` to `MeshBlockPack` and `ParArray1D<Coordinates_t> coord_array` to `MeshBlock`.

File: src/coordinates/get_coordinates.cpp (hunk 2 above)
Comment (Yurlungur) [reply src/coordinates/get_coordinates.cpp:28]: Ah, good question... I'm not sure. The solution of adding a kokkos view to `VariablePack`s would fix that, but incur other overhead. I don't know which solution is more performant. Maybe worth trying them both out.

**Comment by Yurlungur** [comment]
Obviously only one of this or #617 should be merged.

**Comment by Yurlungur** [comment]
I am now in favor of #617 over this PR, so I am closing this one.

**Pull request closed without merging**